src/staged_policy.jl

```diff
@@ -0,0 +1,44 @@
+
+"""
+StagedPolicy
+
+Policy for finite horizon POMDP solvers.
+Wraps multiple Policies, one for each stage.
+"""
+struct StagedPolicy{M<:FiniteHorizonPOMDPs.FHWrapper, P<:Policy}<:Policy
+    pomdp::M
+    staged_policies::Array{P, 1}
+end
+
+
+### functional methods ###
+value(p::StagedPolicy, b, t) = t <= horizon(p.pomdp) ? value(p.staged_policies[t], b) : 0.
+
+action(p::StagedPolicy, b, t) = t <= horizon(p.pomdp) ? action(p.staged_policies[t], b) : ordered_actions(p.pomdp)[1]
+
+# Given that, for example, POMDPSimulators.jl does not work with action(p, b, t),
+# we need to somehow introduce the stage variable
+action(p::StagedPolicy, b::DiscreteBelief) = action(p, b, stage(p.pomdp, b.state_list[1]))
```

Comment:
I think this is correct. To make it a little more picky (which is good), you could do
```suggestion
action(p::StagedPolicy, b::DiscreteBelief) = action(p, b, only(unique(map(stage, p.pomdp, b.state_list))))
```